File: Algorithms/src/3163. String Compression III/Code/solution.java
class Solution {
    public String compressedString(String word) {
        StringBuilder comp = new StringBuilder();
        int count = 1;

        for (int i = 1; i <= word.length(); i++) {
            if (i == word.length() || word.charAt(i) != word.charAt(i - 1) || count == 9) {
                comp.append(count).append(word.charAt(i - 1));
                count = 1;
            } else {
                count++;
            }
        }

        return comp.toString();
    }
}

File: Algorithms/src/3163. String Compression III/Code/solution.cpp
class Solution
{
public:
    string compressedString(string word)
    {
        string comp;
        int count = 1;

        for (int i = 1; i <= word.size(); ++i)
        {
            if (i == word.size() || word[i] != word[i - 1] || count == 9)
            {
                comp += to_string(count) + word[i - 1];
                count = 1;
            }
            else
            {
                count++;
            }
        }

        return comp;
    }
};
